### DISCUSSION TITLE
Google Authentication overrides

### DISCUSSION THREAD

**Discussion by jsalgado10** [OrchardCMS/OrchardCore#7837] [Q&A] [answered]
I am trying to implement google authentication on my CMS page, but need to override a couple of methods and options

1. I need to add "?prompt=consent" to the Authorization end point, so it prompts to select an account when the user tries login again
2. Override the Generateusername method so it extracts the username from the email address

**Comment by jsalgado10** [comment, marked as answer]
Update:
1. I did not need to add "?prompt=consent" to the Authorization end point url as it was not needed anymore
2. I was able to created a new IExternalLoginEventHandler and implement the GenerateUsername method to return the username extracted from the email

**Reply by MichaelPetrinolis** [reply]
You could also use the script in the admin settings to avoid writing a custom module.
